### PR TITLE
Fix #3934: MultiSelect Add selectAllLabel

### LINKED ISSUE #3934: MultiSelect: Add selectAllLabel
### Describe the feature you would like to see added

Hello, 
I propose to add a selectAllLabel props to MultiSelect component.

image

### Is your feature request related to a problem?

I am suggesting this addition because I have personally encountered the need to add this label in one of my projects.
As I did not have the capability to do it natively, I had to use DOM manipulation in javascript to add the label when 'onShow' event is triggered.

### Describe the solution you'd like

I'd like a new optional props (string) on MultiSelect component to define a label for the selectAll checkbox. 
If showSelectAll is true and selectAllLabel is defined, checkbox is display with its label. 

### Describe alternatives you have considered

_No response_

### Additional context

If my proposal is accepted, I would like to work on this issue.

### PR DESCRIPTION
**Pull request by mflagey**
Fix #3934 : MultiSelect Add selectAllLabel

### CODE REVIEW

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/multiselect/MultiSelectHeader.js [hunk 1]
```diff
@@ -61,7 +61,12 @@ export const MultiSelectHeader = React.memo((props) => {
     };
 
     const filterElement = createFilterElement();
-    const checkboxElement = props.showSelectAll && <Checkbox checked={props.selectAll} onChange={onSelectAll} role="checkbox" aria-checked={props.selectAll} />;
+    const checkboxElement = props.showSelectAll && (
+        <div>
+            <Checkbox checked={props.selectAll} onChange={onSelectAll} role="checkbox" aria-checked={props.selectAll} />
+            {props.showSelectAll && !props.filter && <label className="p-multiselect-select-all-label">{props.selectAllLabel}</label>}
+        </div>
```
[the remark this answers is not in the document]
Comment (mflagey) [reply components/lib/multiselect/MultiSelectHeader.js:68]: Something like this ?

```
const checkboxElement = props.showSelectAll && (
        <div>
            <Checkbox id="select-all-checkbox" checked={props.selectAll} onChange={onSelectAll} role="checkbox" aria-checked={props.selectAll} />
            {props.showSelectAll && !props.filter && <label for="select-all-checkbox" className="p-multiselect-select-all-label">{props.selectAllLabel}</label>}
        </div>
);
 ```

[the remark this answers is not in the document]
Comment (melloware) [reply components/lib/multiselect/MultiSelectHeader.js:72]: Yep except for we need to do this...

```js
import { classNames, ObjectUtils, UniqueComponentId } from '../utils/Utils';

const selectAllId = props.id ? props.id + '_selectall' : UniqueComponentId();
```

if you have multiple MultiSelects on the page it has to be unique so you will also have to pass props.id into the Header.

[the remark this answers is not in the document]
Comment (mflagey) [reply components/lib/multiselect/MultiSelectHeader.js:73]: Ok , I changed it!

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
Looking pretty good now!

**Comment by melloware** [comment]
Assigned to PrimeTek to review.

**Comment by mflagey** [comment]
Hello @melloware @mertsincan,
I have rebase my branch. 
It appears that several files, that I hasn't modified, have format errors and make CI fail.
It's probably a prettier check error from master branch

**Comment by melloware** [comment]
Can you rebase this PR a lot has changed since your last commit.

**Pull request merged**